### DISCUSSION TITLE
netbox ansible device creation

### DISCUSSION THREAD

**Discussion by sag2643** [netbox-community/netbox#13861] [General]
Hi all, I am facing issues with the below. Though I mentioned the device_role it always throw me an error. 
msg": "{\"role\":[\"This field is required.\"]}"}
can any one help me

```
    - name: Create device with tags
      netbox.netbox.netbox_device:
        netbox_url: http://netbox.local
        netbox_token: thisIsMyToken
        data:
          name: Another Test Device
          device_type: C9410R
          device_role: Core Switch
          site: Main```

**Comment by abhi1693** [comment]
If you are using NetBox v3.6.0, you need to pass `role` instead of `device_role`. But you should check this with the ansible module whether it has added the support for it.

Please refer to the Breaking changes section of the [release notes](https://github.com/netbox-community/netbox/releases/tag/v3.6.0)

**Reply by sag2643** [reply]
yes i tried with role as well but i still have the same issues. not sure if ansible has added it support 
 "The task includes an option with an undefined variable. The error was: 'dict object' has no attribute 'device_role'. 'dict object' has no attribute 'device_role'\n

**Reply by abhi1693** [reply]
Looks like it has not yet been released https://github.com/netbox-community/ansible_modules/pull/1066

**Reply by sag2643** [reply]
ok thanks for the fast response. so we just have to wait for this to be fixed

**Reply by bitcollector1** [reply]
or stop using Ansible and being dependent on others ;)

**Reply by sag2643** [reply]
really... ?

**Reply by lasag-2511** [reply]
hi @abhi1693 : did it work for you?

**Reply by abhi1693** [reply]
I don't use ansible
